backend/OKPluginVariableRendering/VariableRendering.cs

```diff
@@ -0,0 +1,358 @@
+        }
+
+        const int moduleGroupPrio = 10;
+        const int customersGroupPrio = 100;
+        const int networkSegmentGroupPrio = 1000;
+        const int networkInterfaceGroupPrio = 2000;
+        const int assignmentGroupDefaultPrio = 10000;
+        const int ciPrio = 100000;
+
+        public override async Task<bool> Run(Layer targetLayer, JObject config, IChangesetProxy changesetProxy, IModelContext trans, ILogger logger)
+        {
+
+            logger.LogDebug("Start VariableRendering");
+
+            Configuration cfg;
+
+            try
+            {
+                cfg = config.ToObject<Configuration>();
+                logger.LogDebug("Parsed configuration for VariableRendering.");
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("An error ocurred while creating configuration instance.", ex);
+                return false;
+            }
+
+            var layersetVariableRendering = await layerModel.BuildLayerSet(cfg.InputLayerSet.ToArray(), trans);
+
+            var allCIs = await ciModel.GetMergedCIs(new AllCIIDsSelection(), layersetVariableRendering, false, AllAttributeSelection.Instance, trans, changesetProxy.TimeThreshold);
+
+            var activeTrait = await traitsProvider.GetActiveTrait(cfg.BaseCI.RequiredTrait, trans, changesetProxy.TimeThreshold);
+
+            if (activeTrait == null)
+            {
+                logger.LogError($"Could not find baseCI required trait {cfg.BaseCI.RequiredTrait}");
+                return false;
+            }
+
+            var mainCIs = await effectiveTraitModel.FilterCIsWithTrait(allCIs, activeTrait, layersetVariableRendering, trans, changesetProxy.TimeThreshold);
+
+            mainCIs = mainCIs.Take(10);
+
+            //TODO: select only the realtions that are defined in configuration
+            //      check if selection with specific predicates is possible
+            var relations = new List<string>();
+
+            cfg.BaseCI.FollowRelations.ForEach(r =>
+            {
+                r.Follow.ForEach(f =>
+                {
+                    if (relations.IndexOf(f.Predicate[1..]) == -1)
+                    {
+                        relations.Add(f.Predicate[1..]);
+                    }
+                });
+            });
+
+            var allRelations = new List<Relation>();
+
+            foreach (var layer in cfg.InputLayerSet)
+            {
+                var layerRelations = await relationModel.GetRelations(RelationSelectionAll.Instance, layer, trans, changesetProxy.TimeThreshold);
+
+                allRelations.AddRange(layerRelations);
+            }
+
+
+            var fragments = new List<BulkCIAttributeDataLayerScope.Fragment>();
+
+            foreach (var mainCI in mainCIs)
+            {
+                var mainCIname = new GatheredAttribute { Name = "__name", Value = new AttributeScalarValueText(mainCI.CIName), SourceCIID = mainCI.ID };
```

Comment:
Is there a reason you are treating the __name attribute special? There's no hard requirement that every CI needs a name in every layer.

Reply:
The idea here was to have also name of the CI in this layer but since there is not a requirement that the name to exist also in this layer I have removed it.